doc/developer/entrypoints.rst

```diff
@@ -0,0 +1,82 @@
+Entry Points
+============
+
+**Entry Points** are extensible routines that you can choose
+when you invoke your application. The :py:mod:`built-in entry points <armi.cli>` 
```

Comment:
Maybe something to the tune of "entry points are like the verbs that your App can *do*," and "are invoked by running `> myapp [verb]`"